editor/src/core/shared/dom-utils.ts

```diff
@@ -200,6 +200,7 @@ export const intrinsicHTMLElementNamesThatSupportChildren: Array<string> = [
   'menu',
   'nav',
   'section',
+  'span',
```

Comment:
this _just_ came up in conversation with @maltenuhn that we should probably not allow inserting/reparenting into a span if it only has text. 
If we don't want to build detection for "if it only has text" right now, I would say that as a first approximation, I would simply not allow inserting/reparenting into a span, period. (which seems to be the case on master?)

Reply:
Is this specific to spans or all elements?

Reply:
So it's only not allowing you to insert into a span because of what has been observed earlier which is that intrinsic elements are being parsed into the `Right` of the `Either`, not the `Left` as the original version of the test for spans were expecting. I'd argue it would then be easier to add in this special case after this rather than changing this PR for now.